HomeWork3/Source.cpp

```diff
@@ -0,0 +1,241 @@
+#include <iostream>
+#include <random>
+
+enum ARRAY_FILLING { DEFAULT = 1, RANDOM, CUSTOM };
+char *alloc_char_array(size_t kSize) {
+  char *array = new char[kSize];
+
+  return array;
+}
+void fill_char_array(char *array, size_t kSize) {
+  std::cout << "Please, enter char string with max " << kSize << " elements"
+            << std::endl;
+  std::cin >> array;
+}
+size_t length_char_array(char *array) {
+  if (array == nullptr) {
+    return 0;
+  };
+  size_t arr_size{0};
+  while (array[arr_size] != '\0') {
+    ++arr_size;
+  }
+  return arr_size;
+}
+void CharArrayReverse(char *array, int size) {
+  if (array == nullptr) {
+    return;
+  };
+  if (size != 0) {
+    char *rev = new char[size];
+    for (int i{0}; i < size / 2; ++i) {
+      int tmp{array[size - 1 - i]};
+      array[size - 1 - i] = array[i];
+      array[i] = tmp;
+    }
+
+    delete[] rev;
+  } else {
+    std::cout << "Array is empty." << std::endl;
+  }
+}
+void PrintCharArray(char *array, int size) {
+  if (array == nullptr && size == 0) {
+    std::cout << "Array is empty." << std::endl;
+    return;
+  };
+
+  for (int i{0}; i < size; ++i) {
+    std::cout << array[i];
+  }
+  std::cout << std::endl;
+}
+void PrintUpperCaseCharArray(char *array, int size) {
+  if (array == nullptr) {
+    return;
+  }
+  if (size != 0) {
+    for (int i{0}; i < size; ++i) {
+      if (array[i] >= 'a' && array[i] <= 'z') {
+        std::cout << static_cast<char>(array[i] - 32);
+      } else {
+        std::cout << array[i];
+      }
+    }
+    std::cout << std::endl;
+  } else {
+    std::cout << "Array is empty." << std::endl;
+  }
+}
+void PrintLowerCaseCharArray(char *array, int size) {
+  if (array == nullptr && size == 0) {
+    std::cout << "Array is empty." << std::endl;
+    return;
+  }
+
+  for (int i{0}; i < size; ++i) {
+    if (array[i] >= 'A' && array[i] <= 'Z') {
+      std::cout << static_cast<char>(array[i] + 32);
+    } else {
+      std::cout << array[i];
+    }
+  }
+  std::cout << std::endl;
+}
+void print_is_digit(char *array, int size) {
+  if (array == nullptr && size <= 0) {
+    std::cout << "Array is empty." << std::endl;
+    return;
+  }
+
+  bool digit_found{false};
+  size_t iter{};
+  while (array[iter] != '\0') {
+    if (array[iter] >= '0' && array[iter] <= '9') {
+      digit_found = true;
+      break;
+    }
+    ++iter;
+  }
+  if (digit_found == true) {
+    char *dig_array = new char[size];
+
+    for (int i{0}; i < size; ++i) {
+      if (array[i] >= '0' && array[i] <= '9') {
+        dig_array[i] = array[i];
+      }
+    }
+    if (digit_found) {
+      std::cout << "is_digit: ";
+      size_t digit_iter{};
+      while (dig_array[digit_iter] >= '0' && dig_array[digit_iter] <= '9') {
+        std::cout << dig_array[digit_iter++] << " ";
+      }
```

Comment:
немного не понял этого цикла
почему не так 

```suggestion
      while (dig_array[digit_iter] != '\0') {
        std::cout << dig_array[digit_iter++] << " ";
      }
```

Reply:
dig_array[digit_iter] != '\0') нельзя, в этом массиве нет такого символа(

Reply:
Как нет нуль терминального символа в массиве символов?

Reply:
потому, что это уже новый массив, в который записаны только цифры
 if (digit_found == true) {
    char *dig_array = new char[size];
    size_t iter{};
    for (int i{0}; i < size; ++i) {
      if (array[i] >= '0' && array[i] <= '9') {
        dig_array[iter++] = array[i];
      }
    }

Reply:
> потому, что это уже новый массив, в который записаны только цифры
>  if (digit_found == true) {
>     char *dig_array = new char[size];
>     size_t iter{};
>     for (int i{0}; i < size; ++i) {
>       if (array[i] >= '0' && array[i] <= '9') {
>         dig_array[iter++] = array[i];
>       }
>     }

В таком виде `char *dig_array = new char[size];` не будет. 
А в таком виде будет всегда
`char *dig_array = new char[size]{};`

Reply:
super,
 Вы же точно  рассказывали в начале))